scico/optimize/_pgm.py

```diff
@@ -226,41 +163,21 @@ def __init__(
         L0: float,
         x0: Union[JaxArray, BlockArray],
         step_size: Optional[PGMStepSize] = None,
-        maxiter: int = 100,
-        itstat_options: Optional[dict] = None,
+        **kwargs,
     ):
         r"""
-
         Args:
             f: Loss or Functional object with `grad` defined.
             g: Instance of Functional with defined prox method.
             L0: Initial estimate of Lipschitz constant of f.
             x0: Starting point for :math:`\mb{x}`.
             step_size: helper :class:`StepSize` to estimate the Lipschitz
                 constant of f.
-            maxiter: Maximum number of PGM iterations to perform.
```

Comment:
I miss seeing `maxiter` listed in the docs for each optimizer. What do you think about putting it back (while still refactoring out `itstat_options`)? Or maybe mention it in the help string for `**kwargs`?

Reply:
I see arguments for an against.

For: It would improve readability of individual optimizer documentation, and is a relatively small change.

Against: One of the reasons for use of `**kwargs` and moving the docs for those options to the base class is to avoid duplication of docs, which inevitably leads to errors when changes are made. Keeping the `maxiter` docs in the derived classes would be inconsistent with this principle.

Thoughts?

Reply:
It's a tough call. I value the readability of the docs over the maintenance issue, but not strongly. And, the PR already is as it is, so perhaps we leave it.

Reply:
Let's go with the PR as-is. We can make the suggested changes later if we decide it's a serious issue.